<launch>
  <arg name="ticks_per_meter" value="10000" />
  <arg name="wheel_separation" value="0.2" />
  <arg name="ppr" value="900"/>
  <arg name="d_wheel" value="0.06"/>
  <arg name="wheel_base" value="0.11"/>
  <arg name="urdf_file" default="$(find openbase)/urdf/robot.xacro" />
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg urdf_file)" />
  <arg name="base_frame_id" value="real"/>
  <arg name="odom_frame_id" default="odom"/>
  <node name="$(anon node_odometry)" pkg="openbase" type="node_odometry.py">
    <rosparam subst_value="true">
      base_frame_id: $(arg base_frame_id)
      odom_frame_id: $(arg odom_frame_id)
    </rosparam>
  </node>
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node" output="screen">
    <param name="port" value="/dev/ttyUSB0"/>
  </node>
  <!-- <node name="$(anon node_autonomous)" pkg="openbase" type="node_autonomous.py" output="screen">
    <rosparam subst_value="true">
      base_frame_id: $(arg base_frame_id)
      odom_frame_id: $(arg odom_frame_id)
    </rosparam>
  </node> -->
  <node name="$(anon teleop)" pkg="openbase" type="node_teleop.py" output="screen">
    <rosparam subst_value="true">
      base_frame_id: $(arg base_frame_id)
    </rosparam>
  </node>
</launch>
